### PR TITLE
feat: make it possible to set input as files or dirs to multiqc

### PR DESCRIPTION
**Pull request by Smeds**
make it possible to use the actual input from the rule as input to multiqc, instead of converting it to dirs that are then searched.

### Description
If multiqc search directories it's possible that unwanted files are included to the reports. This will make it possible to only use the provided files, when wanted.

For all wrappers added by this PR, I made sure that

* [x] there is a test case which covers any introduced changes,
* [x] `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* [x] either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* [x] rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* [x] all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* [x] all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* [x] `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command,
* [x] `Snakefile`s pass the linting (`snakemake --lint`),
* [x] `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* [x] Python wrapper scripts are formatted with [black](https://black.readthedocs.io).

### CODE REVIEW

File: bio/multiqc/test/Snakefile [hunk 1]
```diff
@@ -1,10 +1,23 @@
-rule multiqc:
+rule multiqc_dir:
     input:
         expand("samtools_stats/{sample}.txt", sample=["a", "b"])
     output:
         "qc/multiqc.html"
     params:
-        ""  # Optional: extra parameters for multiqc.
+        extrra=""  # Optional: extra parameters for multiqc.
```
Comment (fgvieira) [inline bio/multiqc/test/Snakefile:7]:
```suggestion
        extra=""  # Optional: extra parameters for multiqc.
```

**Comment by fgvieira** [comment]
I find the variable name `convert_file_input_to_dirs` a bit confusing. Could we have something a bit simpler, like (e.g.) `use_input_files_only`, `use_input_dirs`, `scan_input_dirs`?

**Comment by Smeds** [comment]
> I find the variable name `convert_file_input_to_dirs` a bit confusing. Could we have something a bit simpler, like (e.g.) `use_input_files_only`, `use_input_dirs`, `scan_input_dirs`?

good suggestion! I have updated the pull-request with variable name use_input_files_only instead of convert_file_input_to_dirs

**Comment by fgvieira** [comment]
I think it looks goos, but could you also describe this new parameter in the `meta.yaml` file?
Just so it is explained in the documentation.

File: bio/multiqc/meta.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ description: |
 authors:
```
Comment (fgvieira) [inline bio/multiqc/meta.yaml:4]:
```suggestion
url: https://multiqc.info/
authors:
```

**Review by fgvieira — APPROVED** [review]

**Pull request merged**